src/main/java/network/radicle/jetbrains/radiclejetbrainsplugin/config/RadicleSeedNodeDecorator.java

```diff
@@ -0,0 +1,147 @@
+package network.radicle.jetbrains.radiclejetbrainsplugin.config;
+
+import com.intellij.openapi.actionSystem.ActionToolbarPosition;
+import com.intellij.ui.AnActionButton;
+import com.intellij.ui.AnActionButtonRunnable;
+import com.intellij.ui.ToolbarDecorator;
+import com.intellij.ui.table.JBTable;
+import network.radicle.jetbrains.radiclejetbrainsplugin.RadicleBundle;
+import network.radicle.jetbrains.radiclejetbrainsplugin.dialog.AddSeedNodeDialog;
+import network.radicle.jetbrains.radiclejetbrainsplugin.dialog.ConfirmationDialog;
+
+import javax.swing.*;
+import javax.swing.table.DefaultTableModel;
+import java.util.ArrayList;
+import java.util.List;
+
+public class RadicleSeedNodeDecorator {
+
+    private final RadicleSettingsHandler radicleSettingsHandler;
+    private RadicleSettings settings;
+    protected JBTable table;
+    private List<String> loadedSeedNodes;
+    private ArrayList<String> cpLoadedSeedNodes;
+
+    public RadicleSeedNodeDecorator() {
+        this.radicleSettingsHandler = new RadicleSettingsHandler();
+        loadSeedNodes();
+    }
+
+    public void loadSeedNodes() {
+        this.settings = this.radicleSettingsHandler.loadSettings();
+        this.loadedSeedNodes = this.settings.getSeedNodes();
+        cpLoadedSeedNodes = new ArrayList<>(this.loadedSeedNodes);
+    }
+
+    private String concatenate(String seedNode, String port) {
```

Comment:
this string represenation of a seed node - involving a custom separator between host and port is a bit problematic. 

I think it would be best to keep it a separate class (or record, if we had java 17), such as:
```java
public class SeedNode {
  public String host;
  public int port;
...
}
```
and use that to communicate with settings, rather than spreading around the necessary knowledge of how to extract host/port details from a String.